### DISCUSSION TITLE
Regarding libpcap live packet capture support

### DISCUSSION THREAD

**Discussion by maordadush** [microsoft/vcpkg#35395] [Q&A] [answered]
by default, libpcap vcpkg is not compiling the package with live packet capture support. so to support it, the CMake needs to have Packet_ROOT defined before the check-in libpcap's CMakeLists.txt for the packet capture mechanism is done,  how that be accomplished?

I tried to add `-DPacket_ROOT=` argument in the libpcap portfile.cmake file

```
vcpkg_cmake_configure(
    SOURCE_PATH "${SOURCE_PATH}"
    DISABLE_PARALLEL_CONFIGURE
    OPTIONS
        -DPacket_ROOT="C:/npcap-sdk"
        -DDISABLE_NETMAP=ON
        -DDISABLE_BLUETOOTH=ON
        -DDISABLE_DBUS=ON
        -DDISABLE_RDMA=ON
        -DDISABLE_DAG=ON
        -DDISABLE_SEPTEL=ON
        -DDISABLE_SNF=ON
        -DDISABLE_TC=ON
        -DDISABLE_PACKET=OFF
        -DENABLE_REMOTE=OFF
        -DUSE_STATIC_RT=${USE_STATIC_RT}
)
```
and I still get the "live packet capture not supported on this system" error.

**Comment by dg0yt** [comment]
Check (or share) the cmake config log.

**Reply by maordadush** [reply]
@dg0yt 

[config-x64-windows-static-rel-CMakeCache.txt.log](https://github.com/microsoft/vcpkg/files/13510799/config-x64-windows-static-rel-CMakeCache.txt.log)
[config-x64-windows-static-rel-err.log](https://github.com/microsoft/vcpkg/files/13510800/config-x64-windows-static-rel-err.log)
[config-x64-windows-static-rel-out.log](https://github.com/microsoft/vcpkg/files/13510801/config-x64-windows-static-rel-out.log)


Thanks!

**Reply by maordadush** [reply]
@dg0yt 

do you have any idea, please?

**Reply by dg0yt** [reply]
Did you try the changed quote, https://github.com/microsoft/vcpkg/discussions/35395#discussioncomment-7714090?
This is from your CMake cache - guess why:
~~~
//Path to directory with include and lib subdirectories for packet.dll
Packet_ROOT:PATH=C:/vcpkg/buildtrees/libpcap/x64-windows-static-rel/"C:/npcap-sdk"

**Reply by maordadush** [reply]
yes, you can see the change

```
vcpkg_cmake_configure(
    SOURCE_PATH "${SOURCE_PATH}"
    DISABLE_PARALLEL_CONFIGURE
    OPTIONS
        +"-DPacket_ROOT=C:/npcap-sdk"
        -DDISABLE_NETMAP=ON
        -DDISABLE_BLUETOOTH=ON
        -DDISABLE_DBUS=ON
        -DDISABLE_RDMA=ON
        -DDISABLE_DAG=ON
        -DDISABLE_SEPTEL=ON
        -DDISABLE_SNF=ON
        -DDISABLE_TC=ON
        -DDISABLE_PACKET=OFF
        -DENABLE_REMOTE=OFF
        -DUSE_STATIC_RT=${USE_STATIC_RT}
)
```
but then I got 
```

CMake Warning:
  Ignoring extra path from command line:

   "+"-DPacket_ROOT=C:/npcap-sdk""
```
and the Packet_ROOT is empty
```
//Path to directory with include and lib subdirectories for packet.dll
Packet_ROOT:PATH=
```

**Reply by maordadush** [reply]
@dg0yt I also tried without quotation marks at all.

`-DPacket_ROOT=C:/npcap-sdk`

CMake cache:
```
//Path to directory with include and lib subdirectories for packet.dll
Packet_ROOT:PATH=C:/npcap-sdk
```
but then I saw the following errors in my project
![image](https://github.com/microsoft/vcpkg/assets/8192075/5d018951-69c8-4223-a0b9-35125ecba2a4)

**Reply by dg0yt** [reply, marked as answer]
> `"+"-DPacket_ROOT=C:/npcap-sdk""`

This isn't what was meant. The `+` is was part of the diff format (=added line). So either   
`"-DPacket_ROOT=C:/npcap-sdk"`  
or

> `-DPacket_ROOT=C:/npcap-sdk`

---

> but then I saw the following errors in my project

It seems to lack a link library. Check CMake output, or `Packet_LIBRARY` in the cache. I don't know anything about it.

**Comment by dg0yt** [comment]
BTW better change:
~~~diff
--DPacket_ROOT="C:/npcap-sdk"
+"-DPacket_ROOT=C:/npcap-sdk"
~~~
(Rationale: Quote for CMake. CMake will transform for the build tool.)
